jobclient/java/src/main/java/com/twosigma/cook/jobclient/Application.java

```diff
@@ -25,16 +35,44 @@ public String getVersion() {
         return _version;
     }
 
+    public String getWorkloadClass() {
+        return _workloadClass;
+    }
+
+    public String getWorkloadId() {
+        return _workloadId;
+    }
+
+    public String getWorkloadDetails() {
+        return _workloadDetails;
+    }
+
     static JSONObject jsonizeApplication(Application application) throws JSONException {
         final JSONObject object = new JSONObject();
         object.put("name", application._name);
         object.put("version", application._version);
+        if (application._workloadClass != null) {
+            object.put("workload-class", application._workloadClass);
+        }
+        if (application._workloadId != null) {
+            object.put("workload-id", application._workloadId);
+        }
+        if (application._workloadDetails != null) {
+            object.put("workload-details", application._workloadDetails);
+        }
         return object;
     }
 
+    private static String tryGetString(JSONObject object, String key) throws JSONException {
+        return object.has(key) ? object.getString(key) : null;
+    }
```

Comment:
Is there a more "central" place to put this than the `Application` class?

Reply:
I'm ok not putting it somewhere more central - it's very simple. Can re-write as a lambda